modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/IndexStorage.java

```diff
@@ -49,9 +51,29 @@ public interface IndexStorage {
      * <p>Removing a non-existent row is a no-op.
      *
      * @apiNote This method <b>must</b> always be called inside the corresponding partition's
-     *     {@link org.apache.ignite.internal.storage.MvPartitionStorage#runConsistently} closure.
+     *     {@link MvPartitionStorage#runConsistently} closure.
      *
      * @throws StorageException If failed to remove data.
      */
     void remove(IndexRow row) throws StorageException;
+
+    /**
+     * Returns the row ID for which the index needs to be built, {@code null} means that the index building has completed.
+     *
+     * <p>If index building has not started yet, it will return {@link RowId#lowestRowId(int)}.
+     *
+     * @throws StorageException If failed to get the row ID.
+     */
+    @Nullable RowId getNextRowIdToBuild();
```

Comment:
All other methods in this class declare `throws StorageException` explicitly

Reply:
But this exception is not checked, there is no need to specify it, documentation is enough.
I can remove it in other methods, but I think you should not change it.

Reply:
From my side - exception in the signature simply makes it more explicit.
StorageException should have been checked, and we wouldn't have this discussion in such case.

Reply:
We discussed it personally, decided to remake `StorageException` into a checked in the future and therefore add it to the signature.

Reply:
Please don't do that, checked exceptions suck and should not be used